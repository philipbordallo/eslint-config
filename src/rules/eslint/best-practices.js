import { OFF, WARN, ERROR } from 'states';

export default {
  'accessor-pairs': OFF,
  'array-callback-return': [ERROR, {
    allowImplicit: true,
  }],
  'block-scoped-var': ERROR,
  'complexity': OFF,
  'class-methods-use-this': [ERROR, {
    exceptMethods: [],
  }],
  'consistent-return': ERROR,
  'curly': [ERROR, 'multi-line'],
  'default-case': [ERROR, {
    commentPattern: '^no default$',
  }],
  'dot-notation': [ERROR, {
    allowKeywords: true,
  }],
  'dot-location': [ERROR, 'property'],
  'eqeqeq': [ERROR, 'always', {
    null: 'ignore',
  }],
  'guard-for-in': ERROR,
  'max-classes-per-file': [ERROR, 1],
  'no-alert': WARN,
  'no-caller': ERROR,
  'no-case-declarations': ERROR,
  'no-div-regex': OFF,
  'no-else-return': [ERROR, { allowElseIf: false }],
  'no-empty-function': [ERROR, {
    allow: [
      'arrowFunctions',
      'functions',
      'methods',
    ],
  }],
  'no-empty-pattern': ERROR,
  'no-eq-null': OFF,
  'no-eval': ERROR,
  'no-extend-native': ERROR,
  'no-extra-bind': ERROR,
  'no-extra-label': ERROR,
  'no-fallthrough': ERROR,
  'no-floating-decimal': ERROR,
  'no-global-assign': [ERROR, { exceptions: [] }],
  'no-implicit-coercion': [ERROR, {
    boolean: true,
    number: true,
    string: true,
    allow: [],
  }],
  'no-implicit-globals': OFF,
  'no-implied-eval': ERROR,
  'no-invalid-this': OFF,
  'no-iterator': ERROR,
  'no-labels': [ERROR, { allowLoop: false, allowSwitch: false }],
  'no-lone-blocks': ERROR,
  'no-loop-func': ERROR,
  'no-magic-numbers': [OFF, {
    ignore: [],
    ignoreArrayIndexes: true,
    enforceConst: true,
    detectObjects: false,
  }],
  'no-multi-spaces': [ERROR, {
    ignoreEOLComments: false,
  }],
  'no-multi-str': ERROR,
  'no-new': ERROR,
  'no-new-func': ERROR,
  'no-new-wrappers': ERROR,
  'no-octal': ERROR,
  'no-octal-escape': ERROR,
  'no-param-reassign': [ERROR, {
    props: true,
    ignorePropertyModificationsFor: [
      'acc', // for reduce accumulators
      'accumulator', // for reduce accumulators
      'e', // for e.returnvalue
      'ctx', // for Koa routing
      'req', // for Express requests
      'request', // for Express requests
      'res', // for Express responses
      'response', // for Express responses
      '$scope', // for Angular 1 scopes
    ],
  }],
  'no-proto': ERROR,
  'no-redeclare': [ERROR, { builtinGlobals: true }],
  'no-restricted-properties': [
    ERROR,
    {
      object: 'arguments',
      property: 'callee',
      message: 'arguments.callee is deprecated',
    },
    {
      object: 'global',
      property: 'isFinite',
      message: 'Please use Number.isFinite instead',
    },
    {
      object: 'self',
      property: 'isFinite',
      message: 'Please use Number.isFinite instead',
    },
    {
      object: 'window',
      property: 'isFinite',
      message: 'Please use Number.isFinite instead',
    },
    {
      object: 'global',
      property: 'isNaN',
      message: 'Please use Number.isNaN instead',
    },
    {
      object: 'self',
      property: 'isNaN',
      message: 'Please use Number.isNaN instead',
    },
    {
      object: 'window',
      property: 'isNaN',
      message: 'Please use Number.isNaN instead',
    },
    {
      property: '__defineGetter__',
      message: 'Please use Object.defineProperty instead.',
    },
    {
      property: '__defineSetter__',
      message: 'Please use Object.defineProperty instead.',
    },
    {
      object: 'Math',
      property: 'pow',
      message: 'Use the exponentiation operator (**) instead.',
    },
  ],
  'no-return-assign': [ERROR, 'always'],
  'no-return-await': ERROR,
  'no-script-url': ERROR,
  'no-self-assign': [ERROR, {
    props: true,
  }],
  'no-self-compare': ERROR,
  'no-sequences': ERROR,
  'no-throw-literal': ERROR,
  'no-unmodified-loop-condition': ERROR,
  'no-unused-expressions': [ERROR, {
    allowShortCircuit: false,
    allowTernary: false,
    allowTaggedTemplates: false,
  }],
  'no-unused-labels': ERROR,
  'no-useless-call': OFF,
  'no-useless-concat': ERROR,
  'no-useless-escape': ERROR,
  'no-useless-return': ERROR,
  'no-void': ERROR,
  'no-warning-comments': [WARN, {
    terms: ['todo', 'fixme', 'fix', 'xxx'],
    location: 'start',
  }],
  'no-with': ERROR,
  'prefer-promise-reject-errors': [ERROR, { allowEmptyReject: true }],
  'radix': ERROR,
  'require-await': OFF,
  'require-unicode-regexp': OFF,
  'vars-on-top': ERROR,
  'wrap-iife': [ERROR, 'inside', { functionPrototypeMethods: false }],
  'yoda': ERROR,
};
