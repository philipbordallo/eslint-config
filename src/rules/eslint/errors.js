import { WARN, ERROR } from 'src/constants';

export default {
  'for-direction': ERROR,
  'getter-return': [ERROR, { allowImplicit: true }],
  'no-async-promise-executor': ERROR,
  'no-await-in-loop': ERROR,
  'no-compare-neg-zero': ERROR,
  'no-cond-assign': [ERROR, 'always'],
  'no-console': WARN,
  'no-constant-condition': WARN,
  'no-control-regex': ERROR,
  'no-debugger': ERROR,
  'no-dupe-args': ERROR,
  'no-dupe-keys': ERROR,
  'no-duplicate-case': ERROR,
  'no-empty': ERROR,
  'no-empty-character-class': ERROR,
  'no-ex-assign': ERROR,
  'no-extra-boolean-cast': ERROR,
  'no-extra-parens': [ERROR, 'all', {
    conditionalAssign: true,
    nestedBinaryExpressions: false,
    returnAssign: false,
    ignoreJSX: 'all', // delegate to eslint-plugin-react
    enforceForArrowConditionals: false,
  }],
  'no-extra-semi': ERROR,
  'no-func-assign': ERROR,
  'no-inner-declarations': ERROR,
  'no-invalid-regexp': ERROR,
  'no-irregular-whitespace': ERROR,
  'no-loss-of-precision': ERROR,
  'no-misleading-character-class': ERROR,
  'no-obj-calls': ERROR,
  'no-prototype-builtins': ERROR,
  'no-regex-spaces': ERROR,
  'no-sparse-arrays': ERROR,
  'no-template-curly-in-string': ERROR,
  'no-unexpected-multiline': ERROR,
  'no-unreachable': ERROR,
  'no-unsafe-finally': ERROR,
  'no-unsafe-negation': ERROR,
  'require-atomic-updates': ERROR,
  'use-isnan': ERROR,
  'valid-typeof': ERROR,
};
