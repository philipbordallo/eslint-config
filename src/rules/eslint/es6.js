import { OFF, ERROR } from 'states';

export default {
  'arrow-body-style': [ERROR, 'as-needed', {
    requireReturnForObjectLiteral: false,
  }],
  'arrow-parens': [ERROR, 'as-needed', {
    requireForBlockBody: true,
  }],
  'arrow-spacing': [ERROR, {
    before: true,
    after: true,
  }],
  'constructor-super': ERROR,
  'generator-star-spacing': [ERROR, {
    before: false,
    after: true,
    anonymous: 'neither',
    method: {
      before: true,
      after: true,
    },
  }],
  'no-class-assign': ERROR,
  'no-confusing-arrow': [ERROR, { allowParens: true }],
  'no-const-assign': ERROR,
  'no-dupe-class-members': ERROR,
  'no-duplicate-imports': OFF, // replaced by 'import/no-duplicates'
  'no-new-symbol': ERROR,
  'no-restricted-imports': OFF,
  'no-this-before-super': ERROR,
  'no-useless-computed-key': ERROR,
  'no-useless-constructor': ERROR,
  'no-useless-rename': ERROR,
  'no-var': ERROR,
  'object-shorthand': [ERROR, 'always', {
    ignoreConstructors: false,
    avoidQuotes: true,
    avoidExplicitReturnArrows: true,
  }],
  'prefer-arrow-callback': ERROR,
  'prefer-const': [ERROR, {
    destructuring: 'any',
    ignoreReadBeforeAssign: true,
  }],
  'prefer-destructuring': [
    ERROR,
    {
      VariableDeclarator: {
        array: false,
        object: true,
      },
      AssignmentExpression: {
        array: true,
        object: true,
      },
    },
    {
      enforceForRenamedProperties: false,
    },
  ],
  'prefer-numeric-literals': ERROR,
  'prefer-rest-params': ERROR,
  'prefer-spread': ERROR,
  'prefer-template': ERROR,
  'require-yield': ERROR,
  'rest-spread-spacing': [ERROR, 'never'],
  'sort-imports': OFF,
  'symbol-description': ERROR,
  'template-curly-spacing': ERROR,
  'yield-star-spacing': [ERROR, 'after'],
};
