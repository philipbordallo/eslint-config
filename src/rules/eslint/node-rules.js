import { OFF, ERROR } from 'src/constants';

export default {
  'callback-return': OFF,
  'global-require': ERROR,
  'handle-callback-err': OFF,
  'no-buffer-constructor': ERROR,
  'no-mixed-requires': OFF,
  'no-new-require': ERROR,
  'no-path-concat': ERROR,
  'no-process-env': OFF,
  'no-process-exit': OFF,
  'no-restricted-modules': OFF,
  'no-sync': OFF,
};
