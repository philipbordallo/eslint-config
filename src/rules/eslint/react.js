import { ERROR } from '#src/constants.js';

export default {
  'jsx-quotes': [ERROR, 'prefer-double'],
};
