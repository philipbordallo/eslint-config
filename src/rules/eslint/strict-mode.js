import { ERROR } from 'src/states';

export default {
  strict: [ERROR, 'never'],
};
