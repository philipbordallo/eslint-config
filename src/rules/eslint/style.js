import { OFF, WARN, ERROR } from 'src/states';

export default {
  'array-bracket-newline': [ERROR, 'consistent'],
  'array-bracket-spacing': [ERROR, 'never'],
  'array-element-newline': [ERROR, 'consistent'],
  'block-spacing': ERROR,
  'brace-style': [ERROR, 'stroustrup', { allowSingleLine: true }],
  'camelcase': ERROR,
  'capitalized-comments': OFF,
  'comma-dangle': [ERROR, {
    arrays: 'always-multiline',
    objects: 'always-multiline',
    imports: 'always-multiline',
    exports: 'always-multiline',
    functions: 'always-multiline',
  }],
  'comma-spacing': [ERROR, { before: false, after: true }],
  'comma-style': [ERROR, 'last', {
    exceptions: {
      ArrayExpression: false,
      ArrayPattern: false,
      ArrowFunctionExpression: false,
      CallExpression: false,
      FunctionDeclaration: false,
      FunctionExpression: false,
      ImportDeclaration: false,
      ObjectExpression: false,
      ObjectPattern: false,
      VariableDeclaration: false,
      NewExpression: false,
    },
  }],
  'computed-property-spacing': [ERROR, 'never'],
  'consistent-this': OFF,
  'eol-last': [ERROR, 'always'],
  'func-call-spacing': [ERROR, 'never'],
  'func-name-matching': OFF,
  'func-names': [WARN, 'as-needed'],
  'func-style': OFF,
  'function-paren-newline': [ERROR, 'consistent'],
  'id-blacklist': OFF,
  'id-length': OFF,
  'id-match': OFF,
  'implicit-arrow-linebreak': OFF,
  'indent': [ERROR, 2, {
    SwitchCase: 1,
    VariableDeclarator: 1,
    outerIIFEBody: 1,
    MemberExpression: 1,
    FunctionDeclaration: {
      parameters: 1,
      body: 1,
    },
    FunctionExpression: {
      parameters: 1,
      body: 1,
    },
    CallExpression: {
      arguments: 1,
    },
    ArrayExpression: 1,
    ObjectExpression: 1,
    ImportDeclaration: 1,
    flatTernaryExpressions: false,
    ignoreComments: false,
  }],
  'jsx-quotes': OFF, // See rules/eslint/react
  'key-spacing': [ERROR, {
    beforeColon: false,
    afterColon: true,
    mode: 'strict',
  }],
  'keyword-spacing': [ERROR, {
    before: true,
    after: true,
  }],
  'line-comment-position': OFF,
  'linebreak-style': [ERROR, 'unix'],
  'lines-around-comment': OFF,
  'lines-between-class-members': [ERROR, 'always'],
  'max-depth': OFF,
  'max-len': [ERROR, 100, 2, {
    ignoreUrls: true,
    ignoreComments: false,
    ignoreRegExpLiterals: true,
    ignoreStrings: true,
    ignoreTemplateLiterals: true,
  }],
  'max-lines': OFF,
  'max-lines-per-function': OFF,
  'max-nested-callbacks': OFF,
  'max-params': [ERROR, 3],
  'max-statements': [OFF, 10],
  'max-statements-per-line': [OFF, { max: 1 }],
  'multiline-comment-style': OFF,
  'multiline-ternary': OFF,
  'new-cap': [ERROR, {
    newIsCap: true,
    newIsCapExceptions: [],
    capIsNew: false,
    capIsNewExceptions: [],
  }],
  'new-parens': ERROR,
  'newline-per-chained-call': [ERROR, { ignoreChainWithDepth: 2 }],
  'no-array-constructor': ERROR,
  'no-bitwise': ERROR,
  'no-continue': ERROR,
  'no-inline-comments': OFF,
  'no-lonely-if': ERROR,
  'no-mixed-operators': ERROR,
  'no-mixed-spaces-and-tabs': ERROR,
  'no-multi-assign': ERROR,
  'no-multiple-empty-lines': [ERROR, { max: 2, maxEOF: 0 }],
  'no-negated-condition': OFF,
  'no-nested-ternary': ERROR,
  'no-new-object': ERROR,
  'no-plusplus': ERROR,
  'no-restricted-syntax': [
    ERROR,
    {
      selector: 'ForInStatement',
      message: 'for..in loops iterate over the entire prototype chain, which is virtually never what you want. Use Object.{keys,values,entries}, and iterate over the resulting array.',
    },
    {
      selector: 'ForOfStatement',
      message: 'iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations.',
    },
    {
      selector: 'LabeledStatement',
      message: 'Labels are a form of GOTO; using them makes code confusing and hard to maintain and understand.',
    },
    {
      selector: 'WithStatement',
      message: '`with` is disallowed in strict mode because it makes code impossible to predict and optimize.',
    },
  ],
  'no-tabs': ERROR,
  'no-ternary': OFF,
  'no-trailing-spaces': [ERROR, {
    skipBlankLines: false,
    ignoreComments: false,
  }],
  'no-underscore-dangle': [ERROR, {
    allow: [],
    allowAfterThis: false,
    allowAfterSuper: false,
    enforceInMethodNames: false,
  }],
  'no-unneeded-ternary': [ERROR, { defaultAssignment: false }],
  'no-whitespace-before-property': ERROR,
  'nonblock-statement-body-position': [ERROR, 'beside', { overrides: {} }],
  'object-curly-newline': [ERROR, {
    ObjectExpression: { minProperties: 4, multiline: true, consistent: true },
    ObjectPattern: { minProperties: 4, multiline: true, consistent: true },
    ImportDeclaration: { minProperties: 4, multiline: true, consistent: true },
    ExportDeclaration: { minProperties: 4, multiline: true, consistent: true },
  }],
  'object-curly-spacing': [ERROR, 'always'],
  'object-property-newline': [ERROR, {
    allowAllPropertiesOnSameLine: true,
  }],
  'one-var': [ERROR, 'never'],
  'one-var-declaration-per-line': [ERROR, 'always'],
  'operator-assignment': [ERROR, 'always'],
  'operator-linebreak': [ERROR, 'before', { overrides: { '=': 'none' } }],
  'padded-blocks': [ERROR, {
    blocks: 'never',
    classes: 'never',
    switches: 'never',
  }],
  'padding-line-between-statements': OFF,
  'prefer-object-spread': ERROR,
  'quote-props': [ERROR, 'consistent-as-needed'],
  'quotes': [ERROR, 'single'],
  'require-jsdoc': OFF,
  'semi': [ERROR, 'always'],
  'semi-spacing': [ERROR, { before: false, after: true }],
  'semi-style': [ERROR, 'last'],
  'sort-keys': [OFF, 'asc', { caseSensitive: false, natural: true }],
  'sort-vars': OFF,
  'space-before-blocks': ERROR,
  'space-before-function-paren': [ERROR, {
    anonymous: 'never',
    named: 'never',
    asyncArrow: 'always',
  }],
  'space-in-parens': [ERROR, 'never'],
  'space-infix-ops': ERROR,
  'space-unary-ops': [ERROR, {
    words: true,
    nonwords: false,
    overrides: {},
  }],
  'spaced-comment': [ERROR, 'always'],
  'switch-colon-spacing': [ERROR, { after: true, before: false }],
  'template-tag-spacing': [ERROR, 'never'],
  'unicode-bom': [ERROR, 'never'],
  'wrap-regex': OFF,
};
