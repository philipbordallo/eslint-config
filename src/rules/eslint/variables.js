import { OFF, ERROR } from 'src/constants';

import restrictedGlobals from 'eslint-restricted-globals';

export default {
  'init-declarations': OFF,
  'no-delete-var': ERROR,
  'no-label-var': ERROR,
  'no-restricted-globals': [ERROR, 'isFinite', 'isNaN'].concat(restrictedGlobals),
  'no-shadow': ERROR,
  'no-shadow-restricted-names': ERROR,
  'no-undef': ERROR,
  'no-undef-init': ERROR,
  'no-undefined': ERROR,
  'no-unused-vars': [ERROR, {
    vars: 'all',
    args: 'after-used',
    ignoreRestSiblings: true,
  }],
  'no-use-before-define': [ERROR, {
    functions: true,
    classes: true,
    variables: true,
  }],
};
