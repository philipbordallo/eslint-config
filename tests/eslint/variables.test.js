import fs from 'fs';
import { Linter } from 'eslint';

import rules from 'rules/eslint/variables';


const BASE_OPTIONS = {
  env: {
    es6: true,
    browser: true,
    node: true,
  },
  parserOptions: {
    sourceType: 'module',
    ecmaVersion: 2018,
  },
};

const linter = new Linter();

describe('ESLint variables', () => {
  test('should pass', () => {
    const code = fs.readFileSync('./tests/eslint/__fixtures__/variables-pass.js', 'utf8');

    const output = linter.verify(code, {
      ...BASE_OPTIONS,
      rules,
    });

    expect(output).toHaveLength(0);
    expect(output).toMatchSnapshot();
  });

  test('should fail', () => {
    const code = fs.readFileSync('./tests/eslint/__fixtures__/variables-fail.js', 'utf8');

    const output = linter.verify(code, {
      ...BASE_OPTIONS,
      rules,
    });

    expect(output).toMatchSnapshot();
  });
})
