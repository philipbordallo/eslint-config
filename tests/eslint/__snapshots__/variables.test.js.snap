// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ESLint variables should fail 1`] = `
Array [
  Object {
    "column": 5,
    "endColumn": 10,
    "endLine": 2,
    "line": 2,
    "message": "'label' is assigned a value but never used.",
    "nodeType": "Identifier",
    "ruleId": "no-unused-vars",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 14,
    "endLine": 3,
    "line": 3,
    "message": "Found identifier with same name as label.",
    "nodeType": "LabeledStatement",
    "ruleId": "no-label-var",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 9,
    "endLine": 9,
    "line": 9,
    "message": "Unexpected use of 'isFinite'.",
    "nodeType": "Identifier",
    "ruleId": "no-restricted-globals",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 10,
    "line": 10,
    "message": "Unexpected use of 'isNaN'.",
    "nodeType": "Identifier",
    "ruleId": "no-restricted-globals",
    "severity": 2,
  },
  Object {
    "column": 5,
    "endColumn": 11,
    "endLine": 14,
    "line": 14,
    "message": "'shadow' is assigned a value but never used.",
    "nodeType": "Identifier",
    "ruleId": "no-unused-vars",
    "severity": 2,
  },
  Object {
    "column": 10,
    "endColumn": 16,
    "endLine": 16,
    "line": 16,
    "message": "'action' is defined but never used.",
    "nodeType": "Identifier",
    "ruleId": "no-unused-vars",
    "severity": 2,
  },
  Object {
    "column": 7,
    "endColumn": 13,
    "endLine": 17,
    "line": 17,
    "message": "'shadow' is already declared in the upper scope.",
    "nodeType": "Identifier",
    "ruleId": "no-shadow",
    "severity": 2,
  },
  Object {
    "column": 5,
    "endColumn": 13,
    "endLine": 24,
    "line": 24,
    "message": "Shadowing of global property 'Infinity'.",
    "nodeType": "Identifier",
    "ruleId": "no-shadow-restricted-names",
    "severity": 2,
  },
  Object {
    "column": 10,
    "endColumn": 13,
    "endLine": 25,
    "line": 25,
    "message": "Shadowing of global property 'NaN'.",
    "nodeType": "Identifier",
    "ruleId": "no-shadow-restricted-names",
    "severity": 2,
  },
  Object {
    "column": 10,
    "endColumn": 13,
    "endLine": 25,
    "line": 25,
    "message": "'NaN' is defined but never used.",
    "nodeType": "Identifier",
    "ruleId": "no-unused-vars",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 18,
    "endLine": 31,
    "line": 31,
    "message": "'hasNotBeenDefined' is not defined.",
    "messageId": "undef",
    "nodeType": "Identifier",
    "ruleId": "no-undef",
    "severity": 2,
  },
  Object {
    "column": 5,
    "endColumn": 20,
    "endLine": 35,
    "fix": Object {
      "range": Array [
        358,
        370,
      ],
      "text": "",
    },
    "line": 35,
    "message": "It's not necessary to initialize 'bar' to undefined.",
    "nodeType": "VariableDeclarator",
    "ruleId": "no-undef-init",
    "severity": 2,
  },
  Object {
    "column": 11,
    "endColumn": 20,
    "endLine": 35,
    "line": 35,
    "message": "Unexpected use of undefined.",
    "nodeType": "Identifier",
    "ruleId": "no-undefined",
    "severity": 2,
  },
  Object {
    "column": 13,
    "endColumn": 22,
    "endLine": 39,
    "line": 39,
    "message": "Unexpected use of undefined.",
    "nodeType": "Identifier",
    "ruleId": "no-undefined",
    "severity": 2,
  },
  Object {
    "column": 7,
    "endColumn": 14,
    "endLine": 43,
    "line": 43,
    "message": "'message' was used before it was defined.",
    "nodeType": "Identifier",
    "ruleId": "no-use-before-define",
    "severity": 2,
  },
]
`;

exports[`ESLint variables should pass 1`] = `Array []`;
